src/openms/source/QC/TIC.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright The OpenMS Team -- Eberhard Karls University Tuebingen,
+// ETH Zurich, and Freie Universitaet Berlin 2002-2018.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Chris Bielow $
+// $Authors: Tom Waschischeck $
+// --------------------------------------------------------------------------
+
+
+#include <OpenMS/QC/TIC.h>
+#include <OpenMS/FILTERING/TRANSFORMERS/LinearResamplerAlign.h>
+
+using namespace std;
+
+namespace OpenMS
+{
+  /// Constructor
+  TIC::TIC() = default;
+
+  /// Destructor
+  TIC::~TIC() = default;
+
+  /// Reset
+  void TIC::clear()
+  {
+    results_.clear();
+  }
+
+  /// Applies the resampling algorithm, if a bin size i.e a frequency is given.
+  /// Returns the (possibly resampled) TIC of given MSExperiment.
+  void TIC::compute(const MSExperiment &exp, float bin_size)
+  {
+    MSChromatogram tic = exp.getTIC();
+    if (bin_size > 0)
+    {
+      LinearResamplerAlign lra;
+      Param param = lra.getParameters();
+      param.setValue("spacing", bin_size);
+      lra.setParameters(param);
+      lra.raster(tic);
+    }
+    results_.push_back(tic);
+  }
+
+  /// Returns all results calculated with compute.
+  std::vector<MSChromatogram> TIC::getResults() const
+  {
+    return results_;
+  }
+
+  /// Returns required file input.
```

Comment:
describe what the actual requirement is.
No-one wants to dig into source code for that.